runtime/src/iree/io/formats/gguf/gguf_format.c

```diff
@@ -0,0 +1,641 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/io/formats/gguf/gguf_format.h"
+
+#include <ctype.h>
+
+// File format:
+// https://github.com/philpax/ggml/blob/gguf-spec/docs/gguf.md
+//
+// References:
+// https://github.com/ggerganov/ggml/blob/master/src/ggml.c#L21213
+// https://github.com/ggerganov/llama.cpp/blob/master/gguf-py/gguf/gguf.py#L639
```

Comment:
These URLs linking to specific lines are not stable. Convert to permalinks?
![image](https://github.com/openxla/iree/assets/4010439/332ae6e3-7086-448b-bec7-94b02db51057)
e.g. https://github.com/ggerganov/ggml/blob/ff904332ec131bfa3bd45ac552d2f84010d97e02/src/ggml.c#L19971-L19978